src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -181,30 +184,40 @@ private Iterable<SAMSequenceRecord> getSamSequenceRecordsIterable() {
      */
     protected String[] customCommandLineValidation() {
         if (URI == null) {
-            URI = "file:" + referenceSequence.getReferenceFile().getAbsolutePath();
+            if (GATKBucketUtils.isRemoteStorageUrl(referenceSequence.getHtsPath().getURIString())){
+                URI = referenceSequence.getHtsPath().getURIString();
+            } else {
+                URI = "file:" + referenceSequence.getHtsPath().getURIString();
+            }
```

Comment:
Same here, this is the kind of string manipulation (and bug) that plagues the GATK code base - its really hard to preserve the structure of the URI when you do string manipulation like this. So we don't want to reproduce these patterns in Picard.

As it stands, this code will produce a URI string with a duplicate "file:" prefix for local files, since the protocol scheme is already present in the URI string returned by `getURIString()` (i.e., if the reference is a file called "myref.fasta", the value assigned to `URI` resulting from this will be "file:file:///path/to/myref.fasta").

Instead, you can just use:
```suggestion
                URI = referenceSequence.getHtsPath().getURIString();
```
But then the whole enclosing if block (starting in line 187) becomes redundant and can be replaced with one line:
```suggestion
        URI = referenceSequence.getHtsPath().getURIString();
```
There is no need to test whether the file is remote or not - `PicardHtsPath` is always backed by a valid URI that includes a protocol scheme. If the file is local, the protocol scheme will be "file://".

This is a great example of why `PicardHtsPath` and friends (the base classes) are better currency than `File`/`Path`/`String`, etc. The whole idea is to always use PicardHtsPath as the datatype wherever possible, including in function signatures, and then only interconvert to `String`, `Path`, or, in the worst case, `File`, when you absolutely have to, because you're calling some method that you have no control over. In this case, since the `URI` variable is a String, there is no getting around the conversion to String, but thats a terminal operation, and any intermediate processing should be done on the structured type.